src/app/api/create-quiz/route.ts

```diff
@@ -0,0 +1,63 @@
+import {getServerSession} from 'next-auth'
+import {NextResponse} from 'next/server'
+import {authOptions} from '../auth/[...nextauth]/route'
+import {createQuizService} from '@/lib/core/service'
+
+export async function POST(request: Request) {
```

Comment:
From our previous conversation, I think we will be using REST methods for the API, eg:

GET /api/quiz/123
POST /api/quiz
PUT /api/quiz/123
DELETE /api/quiz/123

Reference: https://www.restapitutorial.com/lessons/httpmethods.html